test/types/list_collections.test-d.ts

```diff
@@ -8,38 +8,30 @@ const db = new MongoClient('').db();
 type EitherCollectionInfoResult = CollectionInfo | Pick<CollectionInfo, 'name' | 'type'>;
 
 // We default to the CollectionInfo result type
-expectType<ListCollectionsCursor<Pick<CollectionInfo, 'name' | 'type'> | CollectionInfo>>(
-  db.listCollections()
-);
+expectType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
 // By default it isn't narrowed to either type
 expectNotType<ListCollectionsCursor<Pick<CollectionInfo, 'name' | 'type'>>>(db.listCollections());
-expectNotType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
+expectType<ListCollectionsCursor<CollectionInfo>>(db.listCollections());
```

Comment:
I think these two tests should just be removed, because they are confusing (this one and the one above it), the comment in L12 is no longer relevant